functions/snippets/src/test/java/com/example/functions/SnippetsTests.java

```diff
@@ -293,6 +288,75 @@ public void helloHttp_bodyParamsPost() throws IOException {
     assertThat(responseOut.toString()).isEqualTo("Hello Jane!");
   }
 
+  @Test(expected = RuntimeException.class)
+  public void retryPubsub_handlesRetryMsg() throws IOException {
+    PubSubMessage pubsubMessage = new PubSubMessage();
+    pubsubMessage.setData("{\"retry\": true}");
+
+    new RetryPubSub().accept(pubsubMessage, null);
+
+    String logMessage = logHandler.getStoredLogRecords().get(0).getMessage();
+  }
+
+  @Test
+  public void retryPubsub_handlesStopMsg() throws IOException {
+    PubSubMessage pubsubMessage = new PubSubMessage();
+    pubsubMessage.setData("{\"retry\": false}");
+
+    new RetryPubSub().accept(pubsubMessage, null);
+
+    String logMessage = logHandler.getStoredLogRecords().get(0).getMessage();
+    assertThat("Not retrying...").isEqualTo(logMessage);
+  }
+
+  @Test
+  public void retryPubsub_handlesEmptyMsg() throws IOException {
+    new RetryPubSub().accept(new PubSubMessage(), null);
+
+    String logMessage = logHandler.getStoredLogRecords().get(0).getMessage();
```

Comment:
If this wasn't in a test, I'd be unhappy as it's assuming a lot.  Ok here, however.